DatadogRUM/Sources/RUMMonitor/Monitor.swift

```diff
@@ -135,38 +133,34 @@ internal class Monitor: RUMCommandSubscriber {
     func process(command: RUMCommand) {
         // process command in event context
         featureScope.eventWriteContext { context, writer in
-            self.queue.sync {
-                let transformedCommand = self.transform(command: command)
+            let transformedCommand = self.transform(command: command)
 
-                _ = self.scopes.process(command: transformedCommand, context: context, writer: writer)
+            _ = self.scopes.process(command: transformedCommand, context: context, writer: writer)
 
-                if let debugging = self.debugging {
-                    debugging.debug(applicationScope: self.scopes)
-                }
+            if let debugging = self.debugging {
+                debugging.debug(applicationScope: self.scopes)
             }
         }
 
         // update the core context with rum context
         featureScope.set(
-            baggage: {
-                self.queue.sync { () -> RUMCoreContext? in
-                    let context = self.scopes.activeSession?.viewScopes.last?.context ??
-                                    self.scopes.activeSession?.context ??
-                                    self.scopes.context
-
-                    guard context.sessionID != .nullUUID else {
-                        // if Session was sampled or not yet started
-                        return nil
-                    }
-
-                    return RUMCoreContext(
-                        applicationID: context.rumApplicationID,
-                        sessionID: context.sessionID.rawValue.uuidString.lowercased(),
-                        viewID: context.activeViewID?.rawValue.uuidString.lowercased(),
-                        userActionID: context.activeUserActionID?.rawValue.uuidString.lowercased(),
-                        viewServerTimeOffset: self.scopes.activeSession?.viewScopes.last?.serverTimeOffset
-                    )
+            baggage: { () -> RUMCoreContext? in
+                let context = self.scopes.activeSession?.viewScopes.last?.context ??
```

Comment:
Also I think self access should be weak here

Reply:
✅ done